src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/controller/catalog-volume.tsx

```diff
@@ -0,0 +1,90 @@
+import {ReactController} from '@akeneo-pim-community/legacy-bridge/src/bridge/react';
+import {DependenciesProvider} from '@akeneo-pim-community/legacy-bridge';
+import {ThemeProvider} from 'styled-components';
+import {pimTheme} from 'akeneo-design-system';
+import React from 'react';
+import * as $ from 'jquery';
+import BaseView = require('pimui/js/view/base');
+
+const mediator = require('oro/mediator');
+const featureFlags = require('pim/feature-flags');
+const FormBuilder = require('pim/form-builder');
+const Routing = require('routing');
+const translate = require('oro/translator');
+const Error = require('pim/error');
+
+class CatalogVolumeController extends ReactController {
+  private formPromise: JQueryPromise<BaseView> | null;
+
+  reactElementToMount() {
+    return (
+      <DependenciesProvider>
+        <ThemeProvider theme={pimTheme}>
+          <div>Work in progress...</div>
+        </ThemeProvider>
+      </DependenciesProvider>
+    );
+  }
+
+  routeGuardToUnmount() {
+    return /pim_enrich_catalog_volume_index/;
+  }
+
+  // @ts-ignore to ensure compatibility with old CVM page, return Promise or JQueryPromise
+  renderRoute() {
+    if (featureFlags.isEnabled('control_volume_monitoring_new_page')) {
+      mediator.trigger('pim_menu:highlight:tab', {extension: 'pim-menu-system'});
+      mediator.trigger('pim_menu:highlight:item', {extension: 'pim-menu-system-catalog-volume'});
+
+      return super.renderRoute();
+    }
+
+    // return old CVM
+    this.formPromise = this.renderForm().fail(response => {
+      const message =
+        response && response.responseJSON
+          ? response.responseJSON.message
+          : translate('pim_enrich.entity.fallback.generic_error');
+      const status = response && response.status ? response.status : 500;
+
+      const errorView = new Error(message, status);
+      errorView.setElement(this.$el).render();
+    });
+
+    return this.formPromise;
```

Comment:
We maybe just have to convert the jquery promise into native ES6 promise to avoid the Typescript warning.
Then we can remove our `ts-ignore` comment: https://github.com/akeneo/pim-community-dev/pull/15634/files#diff-cca12bbb1bce4811155d68c9a6552a8719deda2f1bc81093267e0f1544ed4d87R33

```suggestion
    return Promise.resolve(this.formPromise);
```